### PR TITLE
support inclusion toc fallback

### PR DESCRIPTION
**Pull request by OsmondJiang**
#3776

### CODE REVIEW

File: src/docfx/build/toc/TableOfContentsMap.cs [hunk 1]
```diff
@@ -19,13 +19,34 @@ internal class TableOfContentsMap
...
+            parents = new List<Document>();
+            foreach (var (parent, children) in _tocToTocs)
+            {
+                if (children.Contains(toc))
+                {
+                    parents.Add(parent);
+                }
+            }
+
+            return parents.Any();
```
Comment (superyyrrzz) [inline src/docfx/build/toc/TableOfContentsMap.cs]: >.Any() [](start = 26, length = 6)

minor: `.Count`

File: docs/specs/localization.yml [hunk 2]
```diff
@@ -880,3 +880,29 @@ outputs:
...
+  https://github.com/inclusiontoc/a.zh-cn:
+    - files:
+        docs/a/TOC.yml: |
+          - href: a.md
+            name: reference markdown a
+        docs/a/a.md:
+        docs/a/b.md:
+outputs:
+  docs/a/a.json:
+  docs/a/b.json:
```
Comment (yufeih) [inline docs/specs/localization.yml]: why is _a/a.json_ and _a/b.json_ in the output? the fallback is handled at hosting layer.

Comment (OsmondJiang) [reply docs/specs/localization.yml]: a/a.json and a/b.json exists in loc repo...?

File: src/docfx/build/toc/TableOfContentsMapBuilder.cs [hunk 3]
```diff
@@ -17,7 +16,7 @@ internal class TableOfContentsMapBuilder
         /// Tracks toc files which are included by other toc files
         /// Included toc files are excluded when finding nearest toc for an article.
         /// </summary>
-        private readonly ConcurrentHashSet<Document> _referencedTocs = new ConcurrentHashSet<Document>();
+        private readonly ConcurrentDictionary<Document, IEnumerable<Document>> _tocToTocs = new ConcurrentDictionary<Document, IEnumerable<Document>>();
```
[the remark this answers is not in the document]
Comment (OsmondJiang) [reply src/docfx/build/toc/TableOfContentsMapBuilder.cs]: great idea, let me change it.

File: src/docfx/build/toc/TableOfContentsMapBuilder.cs [hunk 4]
```diff
@@ -33,9 +32,14 @@ internal class TableOfContentsMapBuilder
...
             _tocToDocuments.TryAdd(tocFile, referencedDocuments);
-            foreach (var referencedToc in referencedTocs)
+
+            if (referencedTocs != null)
             {
-                _referencedTocs.TryAdd(referencedToc);
+                foreach (var referencedToc in referencedTocs)
+                {
+                    var referencedBy = _referencedTocs.GetOrAdd(referencedToc, new ConcurrentBag<Document>());
+                    referencedBy.Add(tocFile);
```
Comment (superyyrrzz) [inline src/docfx/build/toc/TableOfContentsMapBuilder.cs]: can merge into one line with `AddOrUpdate`

**Pull request merged**